### ISSUE TITLE
issue with backup (pretty new to this)

### ISSUE THREAD

**Issue by stenzilone** [Kaiede/docker-minecraft-bedrock-backup#6]
Hello, I'm pretty new to this, but I have a basic understanding of what's going on. I have a server using OMV that has docker installed on it. I can access docker through Portainer.io. I set up the bedrock server using stacks. The server is running fine. I have been trying to create the backups. I've tried to map everything correctly, but I'm not sure what I am doing wrong. This is the message I am getting in the container logs for the backups container:

time="2021-09-12T18:09:41-06:00" level=debug msg="Using /backups to match uid and gid"
time="2021-09-12T18:09:41-06:00" level=debug msg="Resolved 0 from environment variable UID"
time="2021-09-12T18:09:41-06:00" level=debug msg="Resolved 100 from environment variable GID"
The operation could not be completed. No such file or directory

This is the stack:

version: "2"

services:
minecraftbedrock-server:
image: itzg/minecraft-bedrock-server
container_name: minecraftbedrock-server
network_mode: host
environment:
- PUID=998
- PGID=100
- EULA=TRUE
- VERSION=1.17.11.01
- MEMORY=15G
- SERVER_NAME:Stenzilone
- ONLINE_MODE=false
- LEVEL_NAME=NoMamesWay
- ALLOW_CHEATS=true
- OPS=2535469741104124
- LEVEL_SEED=1595425848
ports:
- 19132:19132/udp
volumes:
- /srv/dev-disk-by-label-DigitalStorage01/MinecraftBedrock:/data
stdin_open: true
restart: unless-stopped

backup:
image: kaiede/minecraft-bedrock-backup
container_name: MinecraftBackup
restart: always
depends_on:
- "minecraftbedrock-server"
environment:
UID: 0
GID: 100
BACKUP_INTERVAL: "1h"
TZ: "America/Denver"
volumes:
- /var/run/docker.sock:/var/run/docker.sock
- /srv/dev-disk-by-label-DigitalStorage01/MinecraftBedrock:/minecraftbedrock-server
- /srv/dev-disk-by-label-DigitalStorage01/MinecraftBackups/backups:/backups
- /srv/dev-disk-by-label-DigitalStorage01/MinecraftBackups/config.json
stdin_open: true

I'm not sure if the UID and GID are correct

Thanks for your help!

**Comment by g6ai** [comment]
It would help if you provide formatted `docker-compose.yml` and `config.json`. But this `/srv/dev-disk-by-label-DigitalStorage01/MinecraftBackups/config.json` looks like the culprit. Shouldn't it be `srv/dev-disk-by-label-DigitalStorage01/MinecraftBackups/backups/config.json`? Could be why you have the "No such file or directory" error.

**Comment by stenzilone** [comment]
I tried that, but it didn't work. I'm not sure where to find the docker-compose.yml file... here is the config.json file:

{
    "servers": {
        "minecraftbedrock-server": “/worlds"
    },
    "trim": {
        "trimDays":   2,
        "keepDays":   14,
        "minKeep":    2
    }
}

**Comment by Kaiede** [comment]
Are you not using docker-compose to configure your backup service? 

That said, here's the issues I'm noticing with what's shared so far:

* config.json is expected to be at `/backups/config.json` from within the backup container by default. Just put the config.json in your backups folder, and don't try to map it in using a volume mapping.
* Inside the config.json, the path to the worlds folder should be `/minecraftbedrock-server/worlds`. The container isn't aware of where the server was mapped in, so it has to be told the full path that was mapped in. 
* I'm not sure why you want to use `root` as your UID? It just needs to be a user with read access to your minecraft server folder, and write access to the backups folder, and is part of the docker group (so it can talk to the docker socket). I highly recommend that over root as it's more secure.

That said, I'll see if I can make some changes this weekend to improve the error messages when config.json cannot be found to make it easier to self-diagnose this sort of problem.

**Comment by stenzilone** [comment]
I don't know if I am using docker-compose... I'm pretty sure I'm not. I set this up on a server running Open Media Vault. I was able to install docker through the OMVExtras. I am editing the docker files through Portainer.io. When I set the minecraft server up, I made a stack which then set up the containers. I wasn't sure what to use for the UID, so I just put 0. I'm not sure what to change it too. I do have a user that has access to those files, but i'm not sure how to add that user in the code. I'm pretty new to this and don't really have a high level of this type of programming... Thanks for helping me!
